### PR TITLE
Event subscriber

### PR DESCRIPTION
**Pull request by auamarto**

### CODE REVIEW

File: src/Exception/FormNotValidException.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class FormNotValidException extends LogicException implements Exception
+{
+    /**
+     * @var Form
+     */
+    protected $form;
+
+    /**
+     * ActionUnavailableException constructor.
+     * @param string $message
```
Comment (GoodVibesDevelopment) [inline src/Exception/FormNotValidException.php]: missing form @param

File: src/Exception/FormNotValidException.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+     * @param int $code
+     * @param Throwable|null $previous
+     */
+    public function __construct(Form $form, $message = "Form Invalid", $code = 0, Throwable $previous = null)
```
Comment (GoodVibesDevelopment) [inline src/Exception/FormNotValidException.php:28]: long line imho

**Review by GoodVibesDevelopment — APPROVED** [review]
unused imports in FormConfigResolverTest and FormConfigResolver.

happy to merge after minor changes

File: src/Exception/FormNotValidException.php (hunk 2 above)
Comment (auamarto) [reply src/Exception/FormNotValidException.php:28]: 105 chars its long but less than 120 and function have 4 arguments.

**Review by patgrudniewski — COMMENTED** [review]
Where is the doc?

File: Resource/doc/FormEventListener.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+and save data to database via 
+[`EntityManagerInterface`](https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/EntityManagerInterface.php)
+
+``FormEventListener`` is registered under name ``opstalent.form.event.listener``
+
+``FormEventListener`` provides following methods:
+
+ - ``public static function getSubscribedEvents(): array``
+ 
+    Function needed by EventSubscriberInterface to register class as listener.
```
Comment (patgrudniewski) [inline Resource/doc/FormEventListener.md]: escape `EventSubscriberInterface`

File: Resource/doc/FormEventListener.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+[`EntityManagerInterface`](https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/EntityManagerInterface.php)
+
+``FormEventListener`` is registered under name ``opstalent.form.event.listener``
+
+``FormEventListener`` provides following methods:
+
+ - ``public static function getSubscribedEvents(): array``
+ 
+    Function needed by EventSubscriberInterface to register class as listener. 
+    Provides array of events that FormEventListener listen to.
```
Comment (patgrudniewski) [inline Resource/doc/FormEventListener.md]: escape `FormEventListener`

File: Resource/doc/FormEventListener.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    
+ - ``public function saveData(FormEvent $event)``
+ 
+    Checks if Form provided by FormEvent is valid and save data to database.
```
Comment (patgrudniewski) [inline Resource/doc/FormEventListener.md]: escape `Form` and `FormEvent`

**Review by patgrudniewski — APPROVED** [review]

**Pull request merged**